src/infra/dex/mod.rs

```diff
@@ -52,6 +52,8 @@ pub enum Error {
     NotFound,
     #[error("rate limited")]
     RateLimited,
+    #[error("unavailable for legal reasons, banned tokens or similar")]
+    UnavailableForLegalReasons,
```

Comment:
This error is extremely specific and so far doesn't really apply to other dex APIs. IMO it makes sense for a generic component (dex API) to mostly rely on generic errors that can be reused across the different implementations.
For example here I think there is an argument to be made that `OrderNotSupported` would cover the error case quite well. Whether it's not supported because a token is deny listed or solvers don't support buy orders doesn't matter too much I think.

Reply:
I thought of putting it under `OrderNotSupported`, but `UnavailableForLegalReasons` is not actually that specific, it's general enough to have it's own http error code 451. So, it could be the case that other DEX-es return this error code in the future. 
But I am fine with both options, if you prefer what you suggested can you create the issue for it?